packages/grafana-ui/src/components/Text/Text.internal.story.tsx

```diff
@@ -39,93 +37,63 @@ const meta: Meta = {
       ],
     },
     truncate: { control: 'boolean' },
+    italic: { control: 'boolean' },
     textAlignment: {
       control: 'select',
       options: ['inherit', 'initial', 'left', 'right', 'center', 'justify', undefined],
     },
   },
+  args: {
+    element: 'h1',
+    variant: undefined,
+    weight: 'light',
+    textAlignment: 'left',
+    truncate: false,
+    italic: false,
+    color: 'primary',
+    children: `This is an example of a Text component`,
+  },
 };
 
-export const Example: StoryFn = () => {
+export const Example: StoryFn = (args) => {
   return (
     <VerticalGroup>
-      <StoryExample name="Header, paragraph, span and legend elements">
-        <H1>h1. Heading</H1>
-        <H2>h2. Heading</H2>
-        <H3>h3. Heading</H3>
-        <H4>h4. Heading</H4>
-        <H5>h5. Heading</H5>
-        <H6>h6. Heading</H6>
-        <P>This is a paragraph</P>
-        <Legend>This is a legend</Legend>
-        <Span>This is a span</Span>
+      <StoryExample name="Header, paragraph and span">
+        <Text {...args} element="h1">
+          This is a header
+        </Text>
+        <Text {...args} element="h3">
+          This is a paragraph that contains
+          <Text color="success" italic>
+            {' '}
+            a span element with different color and style{' '}
+          </Text>
+          but is comprised within the same block text
+        </Text>
       </StoryExample>
     </VerticalGroup>
   );
 };
 Example.parameters = {
   controls: {
-    exclude: ['variant', 'weight', 'textAlignment', 'truncate', 'color', 'children'],
+    exclude: ['element', 'variant', 'weight', 'textAlignment', 'truncate', 'italic', 'color', 'children'],
   },
 };
 
-export const HeadingComponent: StoryFn = (args) => {
-  return (
-    <div style={{ width: '300px' }}>
-      <H1 variant={args.variant} weight={args.weight} textAlignment={args.textAlignment} {...args}>
-        {args.children}
-      </H1>
-    </div>
-  );
-};
-HeadingComponent.args = {
-  variant: undefined,
-  weight: 'light',
-  textAlignment: 'center',
-  truncate: false,
-  color: 'primary',
-  children: 'This is a H1 component',
-};
-
-export const LegendComponent: StoryFn = (args) => {
+export const Basic: StoryFn = (args) => {
```

Comment:
BTW I just tested this story with a `span` and `truncate=true` and it does not truncate, I think because `span` is by default `inline` and that doesn't truncate. Maybe the `truncate` should also set `display: block`? This will obviously break the text if it is surrounded by other text, but I can't think of why we would want to truncate that kind of text.